webui/src/pages/auth/groups/group/members.jsx

```diff
@@ -26,15 +26,41 @@ const GroupMemberList = ({ groupId, after, onPaginate }) => {
     const [refresh, setRefresh] = useState(false);
     const [showAddModal, setShowAddModal] = useState(false);
     const [attachError, setAttachError] = useState(null);
-
+    const [allUsers, setAllUsers] = useState([]);
     const {results, loading, error, nextPage} = useAPIWithPagination(() => {
         return auth.listGroupMembers(groupId, after);
     }, [groupId, after, refresh]);
-
     useEffect(() => {
         setAttachError(null);
     }, [refresh]);
 
+    const setAllUsersFromLakeFS = async () => {
+        if (allUsers.length > 0) {
+            return allUsers
+        }
+        after = ""
+        let hasMore = true
+        let usersList = []
+        try {
+            do {
+                const results = await auth.listUsers("", after, MAX_LISTING_AMOUNT);
+                usersList = usersList.concat(results.results);
+                after = results.pagination.next_offset;
+                hasMore = results.pagination.has_more;
+            } while (hasMore);
+            usersList.sort((a, b) => ResolveEntityDisplayName(a).localeCompare(ResolveEntityDisplayName(b)));
+            setAllUsers(usersList);
+            return usersList;
+        } catch (error) {
+            console.error("Error fetching users:", error);
+            return [];
+        }
+    }
+    const searchUsers = async (prefix, maxResults) => {
+        let allUsersList = await setAllUsersFromLakeFS()
```

Comment:
If I get it right, it fetches all the search results from the API for every prefix change? No caching?
What will happen (performance-wise) in case of 1000 users, for example?

Reply:
No, only on the first time we load the "add member" pop-up